### PR TITLE
ZBUG-1017: update business logic for email address check

### PR DESCRIPTION
**Pull request by ajinkya81**
Made fixes so that the existence of zimbraDataSourceEmailAddress is checked not only createDataSource but also at modifyDataSource

### CODE REVIEW

File: store/src/java/com/zimbra/cs/service/mail/CreateDataSource.java [hunk 1]
```diff
@@ -252,6 +254,21 @@ static void validateFolderId(Account account, Mailbox mbox, Element eDataSource,
             }
         }
     }
+    /**
+     * Confirms that the zimbraDataSourceEmailAddress attribute is unique
+     * 
+     */
+    default static void validateEmailAddr(Account account, Element eDataSource)
```
Comment (chinmay15) [inline store/src/java/com/zimbra/cs/service/mail/CreateDataSource.java]: don't use default access, use public

Comment (dasiyogesh) [inline store/src/java/com/zimbra/cs/service/mail/CreateDataSource.java]: validateEmailAddr() sounds more like email address validation than data source validation, may be changed to validateEmailDataSource()

Comment (ajinkya81) [reply store/src/java/com/zimbra/cs/service/mail/CreateDataSource.java]: fixed

Comment (ajinkya81) [reply store/src/java/com/zimbra/cs/service/mail/CreateDataSource.java]: fixed

File: store/src/java/com/zimbra/cs/service/mail/CreateDataSource.java [hunk 2]
```diff
@@ -252,8 +254,23 @@ static void validateFolderId(Account account, Mailbox mbox, Element eDataSource,
             }
         }
     }
+    /**
+     * Confirms that the zimbraDataSourceEmailAddress attribute is unique
+     * 
+     */
+    public static void validateDataSourceEmail(Account account, Element eDataSource)
```
Comment (jrjena136) [inline store/src/java/com/zimbra/cs/service/mail/CreateDataSource.java]: As this `validateDataSourceEmail(Account account, Element eDataSource)` method is used only for `ModifyDataSource`Class and indirectly called from `processCommonOptionalAttrs()` so its better you can define this method in that class as well. unnecessarily you are creating this in `CreateDataSource` class and calling it from `ModifyDataSource` class

Comment (chinmay15) [reply store/src/java/com/zimbra/cs/service/mail/CreateDataSource.java]: better to move to utility class.

Comment (ajinkya81) [reply store/src/java/com/zimbra/cs/service/mail/CreateDataSource.java]: fixed

File: store/src/java/com/zimbra/cs/service/mail/ModifyDataSource.java [hunk 3]
```diff
@@ -251,4 +254,20 @@ public static void processCommonOptionalAttrs(Map<String, Object> dsAttrs, Eleme
         	dsAttrs.put(Provisioning.A_zimbraDataSourceAttribute, attrList.toArray(new String[0]));
         }
     }
+
+    /**
+     * Confirms that the zimbraDataSourceEmailAddress attribute is unique
+     *
+     */
+    private static void validateDataSourceEmail(Account account, Element eDataSource)
```
Comment (chinmay15) [inline store/src/java/com/zimbra/cs/service/mail/ModifyDataSource.java]: remove static

Comment (chinmay15) [reply store/src/java/com/zimbra/cs/service/mail/ModifyDataSource.java]: my bad. calling method is static

**Comment by ajinkya81** [comment]
rebased & Merged into develop branch.

https://github.com/Zimbra/zm-mailbox/commit/bc4dbb4e7963d05cad96bbb347cfa9a73547428e

**Pull request closed without merging**